Implement fround in JS without depending on Float32Array

arv opened the pull request:
In case Float32Array is not available, use a pure JS
implementation.

This is based on Joshua Bell's code from
https://github.com/inexorabletash/polyfill/blob/master/typedarray.js

Fixes #1703

arv commented:
@johnjbarton PTAL

johnjbarton commented:
LGTM. Our sanity check tests are pretty weak, seems like they may not exercise much code.

arv commented:
PTAL

I made fround.js only export the js implementation and then I had Math.js do the detection of Float32Array. Once that was done I updated the test to test all three paths. (Math.js, fround using Float32Array and fround using js).

I added a few more tests which did catch some bugs. who'd have thunk it

johnjbarton commented:
LTGM.
